gcp-observability/src/main/java/io/grpc/gcp/observability/GcpObservability.java

```diff
@@ -45,23 +69,42 @@ public static synchronized GcpObservability grpcInit() throws IOException {
     if (instance == null) {
       GlobalLoggingTags globalLoggingTags = new GlobalLoggingTags();
       ObservabilityConfigImpl observabilityConfig = ObservabilityConfigImpl.getInstance();
-      Sink sink = new GcpLogSink(observabilityConfig.getDestinationProjectId(),
-          globalLoggingTags.getLocationTags(), globalLoggingTags.getCustomTags(),
-          observabilityConfig.getFlushMessageCount());
+      Sink sink =
+          new GcpLogSink(
+              observabilityConfig.getDestinationProjectId(),
+              globalLoggingTags.getLocationTags(),
+              globalLoggingTags.getCustomTags(),
+              observabilityConfig.getFlushMessageCount());
+      // TODO(dnvindhya): Cleanup code for LoggingChannelProvider and LoggingServerProvider
+      // once ChannelBuilder and ServerBuilder are used
       LogHelper helper = new LogHelper(sink, TimeProvider.SYSTEM_TIME_PROVIDER);
       ConfigFilterHelper configFilterHelper = ConfigFilterHelper.factory(observabilityConfig);
-      instance = grpcInit(sink,
-          new InternalLoggingChannelInterceptor.FactoryImpl(helper, configFilterHelper),
-          new InternalLoggingServerInterceptor.FactoryImpl(helper, configFilterHelper));
+      instance =
+          grpcInit(
+              sink,
+              observabilityConfig,
+              new InternalLoggingChannelInterceptor.FactoryImpl(helper, configFilterHelper),
+              new InternalLoggingServerInterceptor.FactoryImpl(helper, configFilterHelper));
+      instance.registerStackDriverExporter(observabilityConfig.getDestinationProjectId());
     }
     return instance;
   }
 
-  @VisibleForTesting static GcpObservability grpcInit(Sink sink,
+  @VisibleForTesting
+  static GcpObservability grpcInit(
+      Sink sink,
+      ObservabilityConfig config,
       InternalLoggingChannelInterceptor.Factory channelInterceptorFactory,
-      InternalLoggingServerInterceptor.Factory serverInterceptorFactory) {
+      InternalLoggingServerInterceptor.Factory serverInterceptorFactory)
+      throws IOException {
     if (instance == null) {
-      instance = new GcpObservability(sink, channelInterceptorFactory, serverInterceptorFactory);
+      instance =
+          new GcpObservability(sink, config, channelInterceptorFactory, serverInterceptorFactory);
+      LogHelper logHelper = new LogHelper(sink, TimeProvider.SYSTEM_TIME_PROVIDER);
+      ConfigFilterHelper logFilterHelper = ConfigFilterHelper.factory(config);
```

Comment:
Same code as lines 80-81. Can you eliminate 80-81 and just do this?

Reply:
Since lines 80-81 are used by "Logging*Providers" initialization indirectly, I have added a TODO to clean up variables in lines 80-81 while removing "Logging*Providers". And this will be done in the follow up PR.